cert.py

```diff
@@ -144,17 +155,32 @@ def _get_http_challenge(ctx, authzr):
     ctx.fail('no acceptable challenge type found; only HTTP01 supported')
 
 
-def _store_webroot_validation(webroot, challb, val):
+def _store_webroot_validation(ctx, remote, webroot, challb, val):
     logger.info('storing validation of %s' % webroot)
-    try:
-        os.makedirs(os.path.join(webroot, challb.URI_ROOT_PATH), 0o755)
-    except OSError as e:
-        if e.errno != errno.EEXIST:
-            raise
-
-    with open(os.path.join(webroot, challb.path.strip('/')), 'wb') as outf:
-        logger.info('storing validation to %s' % outf.name)
-        outf.write(b(val))
+    chall_path = os.path.join(webroot, challb.path.strip('/'))
+    if not remote:
+        try:
+            os.makedirs(os.path.join(webroot, challb.URI_ROOT_PATH), 0o755)
+        except OSError as e:
+            if e.errno != errno.EEXIST:
+                raise
+
+        with open(chall_path, 'wb') as outf:
+            logger.info('storing validation to %s' % outf.name)
+            outf.write(b(val))
+    else:
+        logger.info('storing validation to %s:%s' % (remote[1], chall_path))
+        ssh = pxssh()
+        ssh.force_password = True
+        ssh.SSH_OPTS = "-o 'NumberOfPasswordPrompts=1'"
+        user, host, port, password = remote
+        try:
+            if ssh.login(host, user, password, port=port):
+                ssh.sendline('echo "%s" > %s' % (b(val), chall_path))
```

Comment:
We might have a slight misunderstanding here: my point was not a vulnerability in the script itself, but the surface area for attack on the server. From the perspective of an administrator installing `wile`, if I wanted to lock down the SSH account used by it, I'd have to know which commands are used and add the appropriate `rbash` settings or `command=(...)` entries to my `authorized_keys`. If we change this command (say, by replacing `echo` with `cat`), we break these setups.
Using something like `paramiko` and its [SFTP support](http://docs.paramiko.org/en/2.1/api/sftp.html) feels like a cleaner way to solve this: you only expect SFTP access because you're only dealing with files. No ambiguities for the admin. Slightly less moving pieces.